subiquity/client/client.py

```diff
@@ -301,7 +325,15 @@ async def _connect():
                 self.client.meta.status.GET(cur=status.state))
         if status.state == ApplicationState.EARLY_COMMANDS:
             print("running early commands")
-            status = await self.client.meta.status.GET(cur=status.state)
+            try:
+                status = await self.client.meta.status.GET(cur=status.state)
+            except aiohttp.client_exceptions.ServerDisconnectedError:
+                # check if the process was already done on non systemd systems
+                # (early commands are blocking there and then, the server is
+                # already shutdown)
+                state = self.get_fallback_server_state()
```

Comment:
Can you help me understand the scenario in which the server process is gone but the client keeps running?  I would have guessed that if the server process was gone, either the client is about to exit normally or something has gone wrong on the server and we need to exit the client with error.

Reply:
I think this about the fact that when the server exits, the last request the client makes doesn't get answered properly so the final server state gets written to a file. It sounds vaguely reasonable to me :)

Reply:
Deprecated since @mwhudson’s branch.